### PR TITLE
feat: update the React Native packages with the new auth system

### PR DESCRIPTION
**Pull request by gdorsi**
Notable changes:
- Renamed BrowserClerkAuth to JazzClerkAuth and renamed the package to jazz-clerk-auth
- move WebSocketPeerWithReconnection to jazz-tools
- unified React Native and Jazz Tools DemoAuth users management
- updated the React Native packages with the new auth system
- updated the RN example apps with the new Auth system

### CODE REVIEW

File: examples/chat-rn/src/chat.tsx [hunk 1]
```diff
@@ -131,6 +137,18 @@ export default function ChatScreen({ navigation }: { navigation: any }) {
...
+            className="rounded h-12 p-2 mb-12 w-40 border border-gray-200 block"
+            value={profile?.name ?? ""}
+            onChangeText={(value) => {
+              if (profile) {
+                profile.name = value;
+              }
+            }}
+            textAlignVertical="center"
+            onSubmitEditing={sendMessage}
+          />
```
Comment (gdorsi) [inline examples/chat-rn/src/chat.tsx:151]: Removed the auth wall and replaced it with a "Username" field

File: packages/jazz-tools/src/implementation/WebSocketPeerWithReconnection.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
+import { Peer } from "cojson";
+import { createWebSocketPeer } from "cojson-transport-ws";
+
+export class WebSocketPeerWithReconnection {
```
Comment (gdorsi) [inline packages/jazz-tools/src/implementation/WebSocketPeerWithReconnection.ts:4]: Refactored the webSocketPeerWithReconnection function into a class because it's easier to manage the internal state this way.

File: packages/jazz-tools/src/tests/syncServer.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import { createServer } from "http";
+import { ControlledAgent, LocalNode } from "cojson";
+import { createWebSocketPeer } from "cojson-transport-ws";
+import { WebSocket, WebSocketServer } from "ws";
+import { TestJSCrypto } from "../testing";
+
+export const startSyncServer = async (port?: number) => {
```
Comment (gdorsi) [inline packages/jazz-tools/src/tests/syncServer.ts:7]: Added a syncServer in the jazz-tools utils to write the tests for WebSocketPeerWithReconnection more easily.

It also helped me on finding some edge cases in the code

File: packages/cojson-transport-ws/src/index.ts [hunk 4]
```diff
@@ -169,6 +173,11 @@ export function createWebSocketPeer({
       return;
     }
 
+    if (!onSuccessCalled) {
+      onSuccess?.();
+      onSuccessCalled = true;
+    }
```
Comment (boorad) [inline packages/cojson-transport-ws/src/index.ts:181]: what if `onSuccess?` ended up undefined?  I guess you "called" it.

File: tests/e2e-react-native/validateCoValue.mjs [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { WebSocket } from "ws";
 
 const invite = process.argv[2];
 
-const [id, inviteSecret] = invite.replace("/invite/", "").split("/");
+const [id, inviteSecret] = invite.replace("#/invite/", "").split("/");
```
Comment (boorad) [inline tests/e2e-react-native/validateCoValue.mjs:7]: yes, I found this little nugget in my PR 😅

File: tests/e2e-react-native/src/App.tsx [hunk 6]
```diff
@@ -3,42 +3,21 @@ import {
...
-    return null;
-  }
 
   return (
     <StrictMode>
-      <JazzProvider
-        auth={auth}
-        peer="wss://cloud.jazz.tools/?key=e2e-rn@garden.co"
-      >
+      <JazzProvider peer="wss://cloud.jazz.tools/?key=e2e-rn@garden.co">
```
Comment (boorad) [inline tests/e2e-react-native/src/App.tsx:20]: the code disapperaring here is so good for DX...

File: packages/cojson-transport-ws/src/index.ts (hunk 4 above)
Comment (gdorsi) [reply packages/cojson-transport-ws/src/index.ts:181]: I can rename the variable in `isFirstMessage` which would make more sense.

Thanks for raising the point!

Also going to add some tests here reusing the syncServer approach.
